### PR TITLE
hr watch all employee data

### PR DESCRIPTION
**Pull request by shivamchouhan123**

### CODE REVIEW

File: app/serializer/api/v1/employee_serializer.rb [hunk 1]
```diff
@@ -3,16 +3,21 @@
...
       
-        attribute :emp_image do |object, params|
-        params[:host] ||= ''
-        if object&.emp_image&.attached?
-          { url: params[:host] + Rails.application.routes.url_helpers.rails_blob_url(
-            object.emp_image, only_path: true
-          ), id: object.emp_image.id }
+        # attribute :emp_image do |object, params|
+        # params[:host] ||= ''
+        # if object&.emp_image&.attached?
```
Comment (ror-mahesh) [inline app/serializer/api/v1/employee_serializer.rb:10]: why you comment this?????

**Review by ror-mahesh — CHANGES_REQUESTED** [review]

File: app/serializer/api/v1/employee_serializer.rb (hunk 1 above)
Comment (shivamchouhan123) [reply app/serializer/api/v1/employee_serializer.rb]: page refresh karo

File: app/controllers/api/v1/employees_controller.rb [hunk 2]
```diff
@@ -7,9 +7,19 @@ class EmployeesController < ApplicationController
       before_action :set_employee, only: %i[show]
       skip_before_action :verify_authenticity_token, only: [:update_password]
 
-      # GET /employees/1 or /employees/1.json
+      def index
+        if current_employee.is_hr?  
+          @employee_data=Employee.all.order(id: :asc)
+          render json: {
```
[the remark this answers is not in the document]
Comment (shivamchouhan123) [reply app/controllers/api/v1/employees_controller.rb:13]: show action also use in @employee_data

[the remark this answers is not in the document]
Comment (ror-mahesh) [reply app/controllers/api/v1/employees_controller.rb:13]: but this is an index action?

**Review by ror-mahesh — APPROVED** [review]
Try to set commit name unique according to your task.

File: app/controllers/api/v1/employees_controller.rb [hunk 3]
```diff
@@ -7,9 +7,19 @@ class EmployeesController < ApplicationController
...
+           message: ['employee profile '], status: 200
+          }
+         end
+      end
+   
+   # GET /employees/1 or /employees/1.json
       def show
-        render json: {
+       render json: {
           data: serializer_data(@employee_data, employee_serializer),
```
Comment (ror-mahesh) [inline app/controllers/api/v1/employees_controller.rb:23]: `@employee_data` not define..

File: app/controllers/api/v1/employees_controller.rb [hunk 4]
```diff
@@ -7,9 +7,19 @@ class EmployeesController < ApplicationController
       before_action :set_employee, only: %i[show]
       skip_before_action :verify_authenticity_token, only: [:update_password]
 
-      # GET /employees/1 or /employees/1.json
+      def index
+        if current_employee.is_hr?
```
Comment (ror-mahesh) [inline app/controllers/api/v1/employees_controller.rb:11]: what would you return when `current_employee` is not HR?

**Review by ror-mahesh — CHANGES_REQUESTED** [review]

File: app/controllers/api/v1/employees_controller.rb (hunk 4 above)
Comment (shivamchouhan123) [reply app/controllers/api/v1/employees_controller.rb:11]: resolve

File: app/controllers/api/v1/employees_controller.rb (hunk 3 above)
Comment (shivamchouhan123) [reply app/controllers/api/v1/employees_controller.rb:23]: resolve

**Review by ror-mahesh — APPROVED** [review]
LGTM

**Pull request merged**